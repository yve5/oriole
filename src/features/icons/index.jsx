import PowerSettingsNew from './components/PowerSettingsNew';
import BoxArrowRight from './components/BoxArrowRight';
import QuestionMark from './components/QuestionMark';
import CircleHalf from './components/CircleHalf';
import Moon from './components/Moon';
import Sun from './components/Sun';

export { BoxArrowRight, CircleHalf, Moon, Sun, QuestionMark, PowerSettingsNew };
