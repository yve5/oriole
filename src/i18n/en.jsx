const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  'About us': 'About us',
  'An error has been detected. Please try again.':
    'An error has been detected. Please try again.',
  Buttons: 'Buttons',
  Changelog: 'Changelog',
  Collapse: 'Collapse',
  Expand: 'Expand',
  Internationalization: 'Internationalization',
  Paging: 'Paging',
  'The request has been sent successfully.':
    'The request has been sent successfully.',
};

export default I18N_DICTIONARY;
