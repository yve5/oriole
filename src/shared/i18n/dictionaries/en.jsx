const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  '$number items per page': '$number items per page',
  '$total records': '$total records',
  Add: 'Add',
  'Ascending order': 'Ascending order',
  Back: 'Back',
  'Changes have been saved.': 'Changes have been saved.',
  'Choose a file': 'Choose a file',
  Columns: 'Columns',
  'Creation date': 'Creation date',
  'Descending order': 'Descending order',
  Disabled: 'Disabled',
  Enabled: 'Enabled',
  First: 'First',
  'Hello World': 'Hello World',
  'Hide columns': 'Hide columns',
  Last: 'Last',
  'Loading...': 'Loading...',
  Members: 'Members',
  Next: 'Next',
  No: 'Non',
  'No record': 'No record',
  'One record': 'One record',
  Previous: 'Previous',
  Reset: 'Reset',
  'This field is invalid.': 'This field is invalid.',
  Title: 'Title',
  'to fill': 'to fill',
  'Update of $update, created at $start by $guest':
    'Update of $update, created at $start by $guest',
  Yes: 'Oui',
};

export default I18N_DICTIONARY;
