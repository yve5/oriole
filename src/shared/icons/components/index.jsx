import PowerSettingsNew from './PowerSettingsNew';
import QuestionMark from './QuestionMark';
import LightMode from './LightMode';
import Contrast from './Contrast';
import DarkMode from './DarkMode';

export { PowerSettingsNew, QuestionMark, LightMode, Contrast, DarkMode };
