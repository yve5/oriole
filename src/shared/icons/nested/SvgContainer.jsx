import React from 'react';
import { clsx } from 'clsx';

const SvgContainer = (props) => {
  const {
    children,
    className = '',
    danger,
    disabled,
    ghost,
    height = '16',
    help,
    info,
    large = false,
    medium = false,
    primary,
    success,
    title = '',
    warning,
    viewBox,
    width = '16',
  } = props;

  const dataTestid = props['data-testid'];

  const classNames = ['svg-icon'];

  if (medium) {
    classNames.push('svg-icon--md');
  } else if (large) {
    classNames.push('svg-icon--lg');
  }

  if (danger) {
    classNames.push('svg-icon--danger');
  } else if (warning) {
    classNames.push('svg-icon--warning');
  } else if (disabled) {
    classNames.push('svg-icon--disabled');
  } else if (primary) {
    classNames.push('svg-icon--primary');
  } else if (success) {
    classNames.push('svg-icon--success');
  } else if (info) {
    classNames.push('svg-icon--info');
  }

  if (ghost) classNames.push('svg-icon--ghost');

  if (help) classNames.push('svg-icon--help');

  return (
    <svg
      width={width}
      height={height}
      fill="currentColor"
      data-testid={dataTestid}
      xmlns="http://www.w3.org/2000/svg"
      viewBox={viewBox || '0 -960 960 960'}
      className={clsx(classNames, className)}
    >
      {children}
      {title && <title>{title}</title>}
    </svg>
  );
};

export default SvgContainer;
