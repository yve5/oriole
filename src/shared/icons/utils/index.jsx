import { getIconComponent } from './Component';

export { getIconComponent };
