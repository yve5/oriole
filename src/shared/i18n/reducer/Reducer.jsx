import en from '../dictionaries/en';
import fr from '../dictionaries/fr';

import { tsl } from '../utils/Translate';
import {
  I18N_CHANGE_LANG,
  I18N_SWITCH_LANG,
  I18N_LOCAL_USER_LANG,
} from '../resources/constants';

const initialState = {
  tsl: (term, args) => tsl(en, term, args),
  dictionaries: [en, fr],
  lang: en.LANG,
};

const reducer = (dictionaries, localStorageName = I18N_LOCAL_USER_LANG) => {
  const getIndex = (dicos, lang) => {
    const index = dicos.findIndex(({ LANG }) => LANG === lang);
    return index === -1 ? 0 : index;
  };

  let innerState = { ...initialState };

  if (Array.isArray(dictionaries) && dictionaries[0]) {
    const subDicos = [];
    dictionaries.forEach((dico) => {
      if (dico.LANG === en.LANG) {
        subDicos.push({ ...en, ...dico });
      } else if (dico.LANG === fr.LANG) {
        subDicos.push({ ...fr, ...dico });
      } else {
        subDicos.push(dico);
      }
    });

    innerState = {
      ...innerState,
      dictionaries: subDicos,
      lang: subDicos[0].LANG,
      tsl: (term, args) => tsl(subDicos[0], term, args),
    };
  }

  const localLang = localStorage.getItem(localStorageName);

  if (localLang) {
    const subDicos = innerState.dictionaries;
    const initialIndex = getIndex(subDicos, localLang);

    innerState = {
      ...innerState,
      lang: localLang,
      tsl: (term, args) => tsl(subDicos[initialIndex], term, args),
    };
  }

  return (state = innerState, action = {}) => {
    switch (action.type) {
      case I18N_CHANGE_LANG: {
        const currentIndex = getIndex(state.dictionaries, action.lang);

        return {
          ...state,
          ...action,
          tsl: (term, args) =>
            tsl(state.dictionaries[currentIndex], term, args),
        };
      }

      case I18N_SWITCH_LANG: {
        const currentIndex = getIndex(state.dictionaries, state.lang);

        const nextIndex = (currentIndex + 1) % state.dictionaries.length;
        const nextDico = state.dictionaries[nextIndex];
        const nextLang = nextDico.LANG;

        localStorage.setItem(localStorageName, nextLang);

        return {
          ...state,
          ...action,
          lang: nextLang,
          tsl: (term, args) => tsl(state.dictionaries[nextIndex], term, args),
        };
      }

      default:
        return state;
    }
  };
};

export default reducer;
