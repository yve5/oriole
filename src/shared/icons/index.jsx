import { getIconComponent } from './utils/Component';
import SvgContainer from './nested/SvgContainer';

export default SvgContainer;
export { getIconComponent };
