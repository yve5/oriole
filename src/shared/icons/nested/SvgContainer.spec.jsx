import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { act, render } from '@testing-library/react';

import reducer from '../../i18n/reducer';
import SvgContainer from './SvgContainer';
import en from '../../i18n/dictionaries/en';
import fr from '../../i18n/dictionaries/fr';

import * as AllIcons from '../../../features/icons';

const mockStore = configureMockStore([thunk]);

const defaultStore = mockStore({ i18n: reducer([fr, en])() });

const DefaultIcon = AllIcons.QuestionMark;

describe('SvgContainer', () => {
  it('should match expected snapshot. default', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <SvgContainer>
          <path d="M200-440v-80h560v80H200Z" />
        </SvgContainer>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. className', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <SvgContainer className="me-5">
          <path d="M200-440v-80h560v80H200Z" />
        </SvgContainer>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. size', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <div>
          <DefaultIcon medium />
          <DefaultIcon large />
        </div>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. color', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <DefaultIcon danger />
        <DefaultIcon disabled />
        <DefaultIcon info />
        <DefaultIcon success />
        <DefaultIcon warning />
        <DefaultIcon primary />
        <DefaultIcon ghost />
        <DefaultIcon help />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. title', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <DefaultIcon title="Hello World" />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. data-testid', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <DefaultIcon data-testid="remove-icon" />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
