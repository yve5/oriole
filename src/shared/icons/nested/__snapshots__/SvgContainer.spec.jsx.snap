// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SvgContainer > should match expected snapshot. color 1`] = `
<DocumentFragment>
  <svg
    class="svg-icon svg-icon--danger"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
  <svg
    class="svg-icon svg-icon--disabled"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
  <svg
    class="svg-icon svg-icon--info"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
  <svg
    class="svg-icon svg-icon--success"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
  <svg
    class="svg-icon svg-icon--warning"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
  <svg
    class="svg-icon svg-icon--primary"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
  <svg
    class="svg-icon svg-icon--ghost"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
  <svg
    class="svg-icon svg-icon--help"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
</DocumentFragment>
`;

exports[`SvgContainer > should match expected snapshot. data-testid 1`] = `
<DocumentFragment>
  <svg
    class="svg-icon"
    data-testid="remove-icon"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
  </svg>
</DocumentFragment>
`;

exports[`SvgContainer > should match expected snapshot. size 1`] = `
<DocumentFragment>
  <div>
    <svg
      class="svg-icon svg-icon--md"
      fill="currentColor"
      height="16"
      viewBox="0 96 960 960"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
        stroke="currentColor"
        stroke-width="12"
      />
    </svg>
    <svg
      class="svg-icon svg-icon--lg"
      fill="currentColor"
      height="16"
      viewBox="0 96 960 960"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
        stroke="currentColor"
        stroke-width="12"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`SvgContainer > should match expected snapshot. title 1`] = `
<DocumentFragment>
  <svg
    class="svg-icon"
    fill="currentColor"
    height="16"
    viewBox="0 96 960 960"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M431 726q1-72 16.5-105t58.5-72q42-38 64.5-70.5T593 409q0-45-30-75t-84-30q-52 0-80 29.5T358 395l-84-37q22-59 74.5-100.5T479 216q100 0 154 55.5T687 405q0 48-20.5 87T601 574q-49 47-59 72t-11 80H431Zm48 250q-29 0-49.5-20.5T409 906q0-29 20.5-49.5T479 836q29 0 49.5 20.5T549 906q0 29-20.5 49.5T479 976Z"
      stroke="currentColor"
      stroke-width="12"
    />
    <title>
      Hello World
    </title>
  </svg>
</DocumentFragment>
`;
